src/main/scala/ai/privado/passes/PropertyParserPass.scala

```diff
@@ -155,30 +160,37 @@ class PropertyParserPass(cpg: Cpg, projectRoot: String, ruleCache: RuleCache, la
   }
 
   private def loadAndConvertYMLtoProperties(file: String): List[(String, String)] = {
-    parser.parse(better.files.File(file).contentAsString) match {
-      case Right(json) => {
-        try {
-          JsonFlattener
-            .flattenAsMap(json.toString)
-            .asScala
-            .toList
-            .map(p => {
-              if (p._2 == null)
-                ("", "")
-              else
-                (p._1, p._2.asInstanceOf[String])
-            })
-        } catch {
-          case e: Throwable =>
-            logger.trace(s"Error while creating properties node for file $file")
-            logger.debug(s"Error while creating properties node for file : $file, error : ${e.getMessage}")
-            List[("", "")]()
-        }
-      }
-      case Left(error) => {
-        List[("", "")]()
+
+    val yamlContent = better.files.File(file).contentAsString // Read the YAML file content as a string
+
+    val yaml                           = new Yaml(new SafeConstructor())
+    val rootNode                       = yaml.compose(new StringReader(yamlContent))
+    var result: List[(String, String)] = List[(String, String)]()
+    processNode(rootNode, "")
+
+    def processNode(node: Node, path: String): Unit = {
+      node match {
+        case mappingNode: MappingNode =>
+          mappingNode.getValue.asScala.foreach { nodeTuple: NodeTuple =>
+            val keyNode   = nodeTuple.getKeyNode.asInstanceOf[ScalarNode]
+            val valueNode = nodeTuple.getValueNode
+            val fullPath  = if (path.isEmpty) keyNode.getValue else s"$path.${keyNode.getValue}"
+            processNode(valueNode, fullPath)
+          }
+        case sequenceNode: SequenceNode =>
+          sequenceNode.getValue.asScala.zipWithIndex.foreach { case (valueNode, index) =>
+            val fullPath = s"$path[$index]"
+            processNode(valueNode, fullPath)
+          }
+        case scalarNode: ScalarNode =>
+          val line   = scalarNode.getStartMark.getLine + 1
+          val column = scalarNode.getStartMark.getColumn + 1
+          val value  = scalarNode.getValue
+          result = result.appended((path, s"$value...$line...$column"))
```

Comment:
instead of concating with `...` and splitting again aftwards, can we use a tuple or some Data Structure?

Reply:
In this case, we'll have to change the structure for all the parsers because we get line numbers for only `yml` at the moment.